File: ios/Plugin/ClipboardPlugin.swift
import Foundation
import Capacitor

@objc(ClipboardPlugin)
public class ClipboardPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ClipboardPlugin"
    public let jsName = "Clipboard"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "read", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "write", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Clipboard()

    @objc func read(_ call: CAPPluginCall) {
        let result = implementation.read()

        if !result.isEmpty {
            call.resolve(result)
        } else {
            call.reject("There is no data on the clipboard")
        }
    }

    @objc func write(_ call: CAPPluginCall) {
        var result: Result<Void, Error>

        if let html = call.options["html"] as? String {
            let text = call.options["text"] as? String
            result = implementation.writeHTML(content: html, text: text ?? "")
        } else if let text = call.options["text"] as? String {
            result = implementation.write(content: text, ofType: Clipboard.ContentType.text)
        } else if let urlString = call.options["url"] as? String {
            result = implementation.write(content: urlString, ofType: Clipboard.ContentType.url)
        } else if let imageBase64 = call.options["image"] as? String {
            result = implementation.write(content: imageBase64, ofType: Clipboard.ContentType.image)
        } else {
            call.reject("No content provided")
            return
        }

        switch result {
        case .success:
            call.resolve()
        case .failure(let err):
            CAPLog.print(err.localizedDescription)
            call.reject(err.localizedDescription)
        }
    }
}

File: android/src/main/java/com/bricksinc/plugins/clipboard/ClipboardPlugin.java
package com.bricksinc.plugins.clipboard;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Clipboard")
public class ClipboardPlugin extends Plugin {

    private Clipboard implementation;

    @Override
    public void load() {
        implementation = new Clipboard(getContext());
    }

    @PluginMethod
    public void write(PluginCall call) {
        String textVal = call.getString("text");
        String htmlVal = call.getString("html");
        String imageVal = call.getString("image");
        String urlVal = call.getString("url");
        String label = call.getString("label");

        ClipboardWriteResponse response;

        if (htmlVal != null) {
            response = implementation.writeHtml(label, textVal != null ? textVal : "", htmlVal);
        } else if (textVal != null) {
            response = implementation.writeText(label, textVal);
        } else if (imageVal != null) {
            response = implementation.writeText(label, imageVal);
        } else if (urlVal != null) {
            response = implementation.writeText(label, urlVal);
        } else {
            call.reject("No data provided");
            return;
        }

        if (response.isSuccess()) {
            call.resolve();
        } else {
            call.reject(response.getErrorMessage());
        }
    }

    @PluginMethod
    public void read(PluginCall call) {
        ClipboardData result = implementation.read();

        if (result == null) {
            call.reject("Unable to read clipboard from the given Context");
        } else if (result.getValue() == null) {
            call.reject("There is no data on the clipboard");
        } else {
            JSObject resultJS = new JSObject();
            resultJS.put("value", result.getValue());
            resultJS.put("type", result.getType());

            call.resolve(resultJS);
        }
    }
}